Branch-SDK/src/io/branch/referral/AnimatedDialog.java

```diff
@@ -0,0 +1,148 @@
+        super(context, cancelable, cancelListener);
+        init(context);
+    }
+
+    //--------------------- Public  methods -------------//
+
+    /**
+     * <p> Opens the dialog with an animation to the content View.</p>
+     */
+    @Override
+    public void show() {
+        slideOpen();
+    }
+
+    /**
+     * <p> Cancels the dialog with an animation to the content View.</p>
+     */
+    @Override
+    public void cancel() {
+        slideClose();
+    }
+
+    /**
+     * </p> Dismiss the dialog.</p>
+     */
+    @Override
+    public void dismiss() {
+        super.dismiss();
+    }
+
+    @Override
+    public void setContentView(int layoutResID) {
+        setDialogWindowAttributes();
+        super.setContentView(layoutResID);
+    }
+
+
+    //------------------Private methods------------------//
+    private void init(Context context) {
+        context_ = context;
+        setDialogWindowAttributes();
+        // Listen for the backpress in order to dismiss the dialog with animation
+        setOnKeyListener(new OnKeyListener() {
+            @Override
+            public boolean onKey(DialogInterface arg0, int keyCode, KeyEvent event) {
+                if (keyCode == KeyEvent.KEYCODE_BACK) {
+                    slideClose();
+                }
+                return true;
+            }
+        });
+    }
+
+    /**
+     * Set the window attributes for the invite dialog.
+     */
+    public void setDialogWindowAttributes() {
+        requestWindowFeature(Window.FEATURE_NO_TITLE);
+        getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
+        getWindow().addFlags(WindowManager.LayoutParams.FLAG_DIM_BEHIND);
+        getWindow().addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
+        WindowManager.LayoutParams lp = new WindowManager.LayoutParams();
+        lp.copyFrom(getWindow().getAttributes());
+        lp.width = WindowManager.LayoutParams.MATCH_PARENT;
+        lp.height = WindowManager.LayoutParams.MATCH_PARENT;
+        lp.gravity = Gravity.BOTTOM;
+        lp.dimAmount = 0.8f;
+        getWindow().setAttributes(lp);
+        getWindow().setWindowAnimations(android.R.anim.slide_in_left);
+        setCanceledOnTouchOutside(true);
+    }
+
+    /**
+     * </p> Opens the dialog with a translation animation to the content view </p>
+     */
+    private void slideOpen() {
+        TranslateAnimation slideUp = new TranslateAnimation(Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, 0, Animation.RELATIVE_TO_SELF, 1.0f, Animation.RELATIVE_TO_SELF, 0f);
+        slideUp.setDuration(500);
+        slideUp.setInterpolator(new AccelerateInterpolator());
+        ((ViewGroup) getWindow().getDecorView()).getChildAt(0).startAnimation(slideUp);
+
+        AnimatedDialog.super.show();
+    }
+
+    /**
+     * </p> Closes the dialog with a translation animation to the content view </p>
+     */
+    private void slideClose() {
+        if (!isClosing_) {
```

Comment:
I'm assuming you're using this to prevent race conditions, but you should do something more like

```
if (isClosing_) {
    return;
}

isClosing = true;
```

Because you have it in that callback, it's possible for slideClose to be called twice before the animation begins, which would still allow this to occur twice.